src/taco-utils/utilHelper.ts

```diff
@@ -219,6 +219,11 @@ module TacoUtility {
         public static expandEnvironmentVariables(str: string): string {
             var regex: RegExp = process.platform === "win32" ? /%(.+?)%/g : /\$(.+?)(?:\/|$)/g;
 
+            // For Mac OS, first try to detect the ~ notation at the start of the string and replace it with the HOME value
+            if (os.platform() === "darwin" && process.env.HOME) {
```

Comment:
I see a lot of if clauses in the code based on the os.platform(). It would be nice if we could have different implementations for the utility classes that use this a lot. installerUtils, utilHelper are good examples.